Bottom: 8b24319740a95873bd89bbaf966befd6f92dc27c
Top:    12f62c20846673de10f7c899de38bc16574a78ad
Author: Peter Clifton <peter@clifton-electronics.co.uk>
Date:   2016-03-05 16:26:16 +0000

Enable output of via pads, and dial down circular approximation accuracy


---

diff --git a/src/hid/common/object3d.c b/src/hid/common/object3d.c
index 7c01b55..a71b461 100644
--- a/src/hid/common/object3d.c
+++ b/src/hid/common/object3d.c
@@ -1447,11 +1447,11 @@ object3d_from_copper_layers_within_area (POLYAREA *area)
     END_LOOP;
 
     fprintf (stderr, "Accumulating pin + via pads\n");
-//    r_search (PCB->Data->pin_tree, &bounds, NULL, pv_copper_callback, &info);
-//    r_search (PCB->Data->via_tree, &bounds, NULL, pv_copper_callback, &info);
+    r_search (PCB->Data->pin_tree, &bounds, NULL, pv_copper_callback, &info);
+    r_search (PCB->Data->via_tree, &bounds, NULL, pv_copper_callback, &info);
 #endif
 
-#if 0
+#if 1
     if (group == top_group ||
         group == bottom_group)
       {
@@ -1461,6 +1461,31 @@ object3d_from_copper_layers_within_area (POLYAREA *area)
       }
 #endif
 
+  /* TODO: Inter-layer features
+   *
+   * Accumulate a circular polygon for each plated hole we may cut, ensuring
+   * the finished polygon contour extends to include the via barrel extents.
+   *
+   * Subtract non-plated hole contours from the polygons.
+   *
+   * For each hole, add the via-barrel between layers... removing the Object3D
+   * from the list of objects as they become joined with some other. (The final
+   * list of objects shuold match 1:1 with resultant bodies, and contain no
+   * duplicates.
+   *
+   * To accomodate overlapping drill holes, accumulate all via-barrels into a
+   * polygon, and subtract that from the positive copper polygon. As we already
+   * added via-barrels to each copper layer, the each barrel extrusion contour
+   * should match only one body of copper on a given layer.
+   *
+   * Remove the drilled hole down each via by extruding the additional faces.
+   *
+   * To accomodate overlapping drill holes, accumulate all drills into a polygon,
+   * and subtract that from the positive copper polygon. Any subtracted contour
+   * should at this point match to one body of copper on a given layer.
+   *
+   */
+
   if (info.poly == NULL)
     {
       fprintf (stderr, "Skipping layer group %i, info.poly was NULL\n", group);
